modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,326 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.modernizer_annotation_processor;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.ArrayType;
+import javax.lang.model.type.ExecutableType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.TypeVisitor;
+import javax.lang.model.util.SimpleTypeVisitor6;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import com.google.auto.service.AutoService;
+import com.google.common.base.Strings;
+
+@SupportedAnnotationTypes("java.lang.SuppressWarnings")
+@AutoService(Processor.class)
+public class ModernizerAnnotationProcessor extends AbstractProcessor {
+
+    @Override
+    public final SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latest();
+    }
+
+    private class AnnotatedElements {
+        private List<String> annotatedClasses;
+        private List<String> annotatedMethods;
+
+        AnnotatedElements(
+            List<String> annotatedClasses,
+            List<String> annotatedMethods
+        ) {
+            this.annotatedClasses = annotatedClasses;
+            this.annotatedMethods = annotatedMethods;
+        }
+
+        public List<String> getAnnotatedClasses() {
+            return annotatedClasses;
+        }
+
+        public List<String> getAnnotatedMethods() {
+            return annotatedMethods;
+        }
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations,
+        RoundEnvironment roundEnv
+    ) {
+        for (TypeElement annotation : annotations) {
+            AnnotatedElements annotatedElements =
+                getAnnotatedElements(roundEnv, annotation);
+            if (!(annotatedElements.getAnnotatedClasses().isEmpty() &&
```

Comment:
While this means the same thing, it might read better as `!isEmpty || !isEmpty`.